src/main/scala/de/heikoseeberger/sbtheader/HeaderCreator.scala

```diff
@@ -64,20 +66,22 @@ final class HeaderCreator private (fileType: FileType,
     }
 
   private val headerNewLine =
-    headerText match {
+    hText match {
       case crlf(_) => "\r\n"
       case cr(_)   => "\r"
       case _       => "\n"
     }
 
-  private val newHeaderText = headerText.replace(headerNewLine, fileNewLine)
+  private val newHeaderText = hText.replace(headerNewLine, fileNewLine)
+  private val hTextNl       = hText + newLine // the regex always checks for a new line at the end of a header text
+  private val nl            = (if (headerEmptyLine) newLine else "")
 
   private val modifiedText =
     text match {
-      case headerPattern(`headerText`, _) => None
-      case headerPattern(_, body)         => Some(firstLine + newHeaderText + body)
-      case body if body.isEmpty           => None
-      case body                           => Some(firstLine + newHeaderText + body.replaceAll("""^\s+""", "")) // Trim left
+      case headerPattern(`hText` | `hTextNl`, _) => None
```

Comment:
This looks complicated, because `commentStyle.pattern` is unaware of the optional new line. How about we change `commentStyle.pattern : RegEx` to `commentStyle.pattern(emptyLine: Boolean) : RegEx` so that the RegEx returned will do this matching for us?

@hseeberger WDYT?